crypto/internal.h

```diff
@@ -893,7 +894,35 @@ static inline void CRYPTO_store_word_le(void *out, crypto_word_t v) {
 }
 
 
+enum fips_approved_algorithm_t {
```

Comment:
Should this be defined in `crypto/internal.h`? If consumers are attempting to write a uniform code-base with both fips and non-fips, they have to do conditionals every time not in fips mode because these enums would be hidden.